code/modules/urist/modules/GOAI/basics/_interactions.dm

```diff
@@ -223,4 +221,16 @@ GLOBAL_LIST_EMPTY(goai_interaction_holders)
 			if(log_on_missing)
 				to_world_log("Failed to get interaction data for [src] - no interaction data!")
 
+		//Climbable atoms are -everywhere- and would require SO many unique holders. This will ensure *all* climbable atoms have a climb interaction, either their own as defined above, or the generic one appended to their holder or in a newly generated one.
+		if(src.atom_flags & ATOM_FLAG_CLIMBABLE)
+			if(!holder)
+				holder = GenerateGenericInteractions(src, /datum/interactions_holder/ClimbHolder)
+				src.interactions_holder = holder
+			else if(!holder.GetAction(null, ACT_CLIMB))
+				var/datum/interaction/climb = GLOB.goai_interactions[/datum/interaction/GenericClimb]
+				if(!climb)
+					climb = new /datum/interaction/GenericClimb
```

Comment:
Hmm, not sure if new()s all over the place will hold up architecturally. That might be a lot of new objects to allocate...

Might have to implement some kind of pooling mechanism for these kinds of common cases so that we could have a ~single instance of the interaction that can take a reference to the bound atom or something...

Reply:
These new()s are all GLOB don't forget. So it shouldn't be _that_ bad. Each `interactions_holder` is a single global instance in an assoc list linked to an atom path, the atom then references it. Each `interaction` is _also_ a global instance that all the various holders then point to. So at any time there'll only ever be a single `/datum/interaction/GenericClimb`, and one `/datum/interactions_holder/ClimbHolder` per type path _if_ it requires it (ie doesn't have it's own holder to overload it) & has been 'touched' by the pathfinder. Both of these assoc lists are checked first before spawning a new datum (holder first, then it checks if its contained interactions are already spawned & points to them).

If that's still too many, I could easily implement a new GLOB list for generic holders and key it by generic action type, rather than owner type path? It wouldn't be unique per type, but shouldn't be necessary for a generic holder.